framework/src/vectorpostprocessors/PointSamplerBase.C

```diff
@@ -161,6 +167,10 @@ PointSamplerBase::getLocalElemContainingPoint(const Point & p)
 {
   const Elem * elem = (*_pl)(p);
 
+  if (_discontinuous_variables)
+    if (_pl->locate_node(p))
```

Comment:
Is this an expensive operation? If so, perhaps we should make this check happen only in debugging modes? And then I guess I could go either way on whether to warn or error. Probably warning makes sense because I could see accidentally falling on top of a node quite often.

Reply:
You could actually just replace this with a check on if the point is contained within any of the sides of the elem. Right now - it's doing a point check, and then another point check. 

```
if (elem && _discontinuous_variables)
  for (const auto s : elem->side_index_range())
    if (elem->build_side_ptr(s)->contains_point(p))
    {
      // your warning...
      break;
    }

Reply:
Sides is a good idea. I settled on node to just fix the issue at hand but this is better.

Reply:
well a little worried about line samplers breaking in all the tests with faces. We ll see.